runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -132,6 +133,9 @@ def __init__(
         self._namespace = kwargs.get("namespace")
         self._context = kwargs.get("context")
 
+        if not dryrun and self.launcher_type == "den" and self.rns_address:
+            DenLauncher.check_status(self)
+
         # Checks if state info is in local sky db, populates if so.
         if not dryrun and not self.ips and not self.creds_values:
```

Comment:
i think this check would also only be relevant if launcher type is "local"? since it relies on a local sky DB